src/controllers/task.js

```diff
@@ -33,49 +12,31 @@ const TaskController = (taskModel) => {
             });
         }
         return res.status(200).json({
-            data: task
+            data: task,
+            error: err ? err.message : ""
         });
     });
 
-    // /event POST
-    /* 
-    Parameters:
-        {
-            "name": string,
-            "description": string,
-            "duration": int
-        }
-    Returns:
-        Success - 200 Status Code
-        {
-            "data": {
-                "id": int
-            }
-        }
-        Error - 400 Status Code
-        {
-            "message": string
-        }
-    */
     router.post('/', async (req, res) => {
         if (!req.body)
             return res.status(400).json({
-                message: "Malformed Request"
+                error: "Malformed Request"
             });
         const body = req.body;
+        if (!("name" in body))
+            return res.status(400).json({
+                error: "Malformed Request"
+            })
+
         const name = body.name;
         const description = body.description;
         const duration = body.duration;
         const [data, err] = await taskModel.createTask(name, description, duration);
```

Comment:
Instead, just check if "name" or "duration" is undefined after getting it from the body (duration should definitely be a required field)